src/test/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActorTest.scala

```diff
@@ -36,9 +37,7 @@ class TaskReplaceActorTest extends AkkaUnitTest with Eventually {
       val instanceA = f.runningInstance(app)
       val instanceB = f.runningInstance(app)
 
-      f.tracker.specInstancesSync(app.id) returns Seq(instanceA, instanceB)
-      f.tracker.get(instanceA.instanceId) returns Future.successful(Some(instanceA))
-      f.tracker.get(instanceB.instanceId) returns Future.successful(Some(instanceB))
+      f.populateInstances(Seq(instanceA, instanceB))
```

Comment:
As I mention further below I question the value in such helper methods.

Reply:
Yikes, me too. That's a quick ticket to mocking and stub hell right there.

Looking through InstaceTrackerDelegate, there are many more abstract methods than required. If we provided a default implementation for the majority of the methods, then special implementations would only need to define 4:

- instancesBySpec
- instance
- process
- instanceUpdates

We could pretty easily define a "real-ish implementation" that is backed by a simple mutable map. This would require far less mocking and stubbing. We could just throw NotImplementedError for populate and instanceUpdates; as far as TaskReplaceActor is concerned, these aren't even used. Alternatively, we could break apart the interfaces and have InstanceTrackerReader and InstanceTrackerWriter.

I'm inclined to believe that such an approach would be easier to grok.